src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -356,14 +356,17 @@ private function registerTemplatingConfiguration(array $config, $ide, ContainerB
         $loader->load('templating.xml');
         $loader->load('templating_php.xml');
 
-        $links = array(
-            'textmate' => 'txmt://open?url=file://%%f&line=%%l',
-            'macvim' => 'mvim://open?url=file://%%f&line=%%l',
-            'emacs' => 'emacs://open?url=file://%%f&line=%%l',
-            'sublime' => 'subl://open?url=file://%%f&line=%%l',
-        );
+        if (!$container->hasParameter('templating.helper.code.file_link_format')) {
+            $links = array(
+                'textmate' => 'txmt://open?url=file://%%f&line=%%l',
+                'macvim' => 'mvim://open?url=file://%%f&line=%%l',
+                'emacs' => 'emacs://open?url=file://%%f&line=%%l',
+                'sublime' => 'subl://open?url=file://%%f&line=%%l',
+            );
 
-        $container->setParameter('templating.helper.code.file_link_format', isset($links[$ide]) ? $links[$ide] : $ide);
+            $link = ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format') ?: $ide;
+            $container->setParameter('templating.helper.code.file_link_format', isset($links[$link]) ? $links[$link] : $link);
```

Comment:
checking ``$links[$link]`` for values coming from XDebug config does not make sense IMO